tools/src/format.rs

```diff
@@ -310,17 +310,72 @@ pub fn format_file_data(cfg: &config::Config,
         let lineno = i + 1;
 
         let blame_data = if let Some(ref lines) = blame_lines {
-            let blame_line = lines[i as usize];
+            let blame_line = &lines[i as usize];
             let pieces = blame_line.splitn(4, ':').collect::<Vec<_>>();
-            let revs = pieces[0];
-            let filespecs = pieces[1];
-            let blame_linenos = pieces[2];
 
-            // TODO: append with comma-separation to revs, filespecs, and blame_linenos
-            // if we want to have multiple blame entries
+            // These store the final data we ship to the front-end.
+            // Each of these is a comma-separated list with one element
+            // for each blame entry.
+            let mut revs = String::from(pieces[0]);
+            let mut filespecs = String::from(pieces[1]);
+            let mut blame_linenos = String::from(pieces[2]);
+
+            if let Some(ref git) = tree_config.git {
+                // These are the inputs to the find_prev_blame operation,
+                // updated per iteration of the loop.
+                let mut cur_rev = pieces[0].to_string();
+                let mut cur_path = PathBuf::from(if pieces[1] == "%" { path } else { pieces[1] });
+                let mut cur_lineno = pieces[2].parse::<u32>().unwrap();
+
+                let mut max_ignored_allowed = 5;  // chosen arbitrarily
```

Comment:
I just didn't want it accidentally go off into the weeds. Also note that the line mapping thing is approximate (with larger changes it may not map to the right line) and so the deeper you go the more the errors may compound. Plus having a huge list of entries in the frontend would probably be unweildy. We don't currently hit this limit, it's just a failsafe kind of thing.